docs/plugin/messaging.md

```diff
@@ -0,0 +1,601 @@
+
+```c++
+void DirectOutput::Output(const Core::Messaging::MessageInfo& messageInfo, const Core::Messaging::IEvent* message) const
+{
+    ASSERT(message != nullptr);
+    ASSERT(messageInfo.Type() != Core::Messaging::Metadata::type::INVALID);
+
+    string result = messageInfo.ToString(_abbreviate).c_str() +
+                    Core::Format("%s\n", message->Data().c_str());
+
+#ifndef __WINDOWS__
+    if (_isSyslog == true) {
+        //use longer messages for syslog
+        syslog(LOG_NOTICE, "%s\n", result.c_str());
+    }
+    else
+#endif
+    {
+        std::cout << result << std::endl;
+    }
+}
+```
+
+# Logging
+
+Logging, on the other hand, is the process of recording events that occur during the execution of an application. These events could be error messages, warnings, or informational messages that provide details about the application’s behavior. Logging is typically used for debugging and troubleshooting purposes. The primary goal of logging is to provide a historical record of events that can be used to analyze and diagnose problems. This means that in theory, we want to save the logs for later use.
+
+```c++
+SYSLOG(Logging::Startup, (_T("Failure in setting Key:Value:[%s]:[%s]\n"), index.Current().Key.Value().c_str(), index.Current().Value.Value().c_str()));
+```
+
+This relates to the main difference between the `TRACE` and `SYSLOG` macros in Thunder, which is that `SYSLOG` is present in any build, and `TRACE` is dropped in production, the same way as, for example, an `ASSERT` macro. It concludes the main distinction between these two message categories in Thunder: tracing should be used when we want to indicate a vital information during the development, and we should use logging to record any important data that could be useful in the future.
+
+```c++
+SYSLOG_GLOBAL(Logging::Fatal, (_T("Plugin config file [%s] could not be opened."), file.Name().c_str()));
+```
+
+Similarly as in the case of tracing, there is also a global version of the `SYSLOG` macro, and the difference is exactly the same as with tracing macros. Furthermore, the `DirectOutput` class can be configured using the `Mode()` method to send the messages to the system logger instead of simply printing them on the console.
```

Comment:
Once again a very valid point, because the `SYSLOG` macro should be used, and we will even mark `SYSLOG_GLOBAL` as deprecated, because that's no longer necessary since the separation of logging and tracing (logging does not have either a class or a function name any more). I'll make changes to this part of the document.